### PR TITLE
Remove codecov upload

### PR DESCRIPTION
**Pull request by La0**

### CODE REVIEW

File: src/codecoverage/bot/code_coverage_bot/codecov.py [hunk 1]
```diff
@@ -58,7 +58,8 @@ def __init__(self, repository, revision, task_name_filter, cache_root, client_id
 
         if revision is None:
             # Retrieve revision of latest codecov build
-            self.github_revision = uploader.get_latest_codecov()
+            # TODO: use HGMO
+            self.github_revision = None
```
Comment (marco-c) [inline src/codecoverage/bot/code_coverage_bot/codecov.py]: We can probably remove this altogether

Comment (La0) [inline src/codecoverage/bot/code_coverage_bot/codecov.py]: @marco-c Can we use HGMO directly here, or we need to find a code coverage build from Taskcluster ?

Comment (marco-c) [reply src/codecoverage/bot/code_coverage_bot/codecov.py]: (We needed git/github only for codecov/coveralls)

Comment (marco-c) [reply src/codecoverage/bot/code_coverage_bot/codecov.py]: We need to get the latest successful build, so we can probably use the backend here (with https://coverage.moz.tools/v2/latest).

Comment (marco-c) [reply src/codecoverage/bot/code_coverage_bot/codecov.py]: (but the github stuff can be removed altogether, you can just set self.revision = latest_successful_revision

File: src/codecoverage/bot/code_coverage_bot/codecov.py [hunk 2]
```diff
@@ -132,9 +133,6 @@ def generate_covdir(self):
     def go_from_trigger_mozilla_central(self):
         commit_sha = self.githubUtils.mercurial_to_git(self.revision)
         try:
```
Comment (marco-c) [inline src/codecoverage/bot/code_coverage_bot/codecov.py]: We can simplify this part, only return when gcp_cov_dir_exists

File: src/codecoverage/bot/tools/covdir_gen.py [hunk 3]
```diff
@@ -59,7 +53,8 @@ def list_commits(maximum=None, unique=None, skip_commits=[]):
             dates.add(week)
 
             # Convert git to mercurial revision
-            commit['mercurial'] = github.git_to_mercurial(commit['commitid'])
+            # TODO: use releng mapper here
+            commit['mercurial'] = None  # convert(commit['commitid'])
```
Comment (marco-c) [inline src/codecoverage/bot/tools/covdir_gen.py]: You can use libmozdata new module

Comment (La0) [reply src/codecoverage/bot/tools/covdir_gen.py]: Done, thanks for your implementation on libmozdata :smiley:

**Comment by La0** [comment]
Rebased on top of master after the cli_common removal

**Pull request merged**